### ISSUE TITLE
`jnp.asarray(HashableArray(jnp.ndarray))` fails

### ISSUE THREAD

**Issue by attila-i-szabo** [netket/netket#1460]
The following MNWE fails:
```python3
a = jnp.asarray([1.0])
b = nk.utils.HashableArray(a)
c = jnp.asarray(b)
```
yielding
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/attila/.local/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 2028, in asarray
    return array(a, dtype=dtype, copy=False, order=order)  # type: ignore
  File "/home/attila/.local/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 2007, in array
    raise TypeError(f"Unexpected input type for array: {type(object)}")
TypeError: Unexpected input type for array: <class 'netket.utils.array.HashableArray'>
```
The odd thing is that there are calls of `jnp.asarray(HashableArray)` in the upstream NetKet code (e.g., [here](https://github.com/netket/netket/blob/23b5cb7c26ee86b179cb374e0ce6158d08d1468d/netket/models/equivariant.py#L145)) that run without problem. Do we understand what determines whether this works or not?

**Comment by PhilipVinc** [comment]
```python
a = np.asarray([1.0])
b = nk.utils.HashableArray(a)
c = jnp.asarray(b)
```

works. I think something breaks for a `jnp.array` wrapped in the hashablearray

**Comment by PhilipVinc** [comment]
closed by #1461
